conftest.py

```diff
@@ -146,19 +147,32 @@ def pytest_sessionfinish(session: pytest.Session, exitstatus):  # pylint: disabl
 def default_configuration():
     """Return the default test compilation configuration."""
 
-    # Remove parameter_selection_strategy once it is set to multi-parameter in Concrete Python
-    # by default
-    # FIXME: https://github.com/zama-ai/concrete-ml-internal/issues/3860
-    # Parameter `enable_unsafe_features` and `use_insecure_key_cache` are needed in order to be
-    # able to cache generated keys through `insecure_key_cache_location`. As the name suggests,
-    # these parameters are unsafe and should only be used for debugging in development
-    return Configuration(
-        dump_artifacts_on_unexpected_failures=False,
-        enable_unsafe_features=True,
-        use_insecure_key_cache=True,
-        insecure_key_cache_location="ConcreteNumpyKeyCache",
-        parameter_selection_strategy=ParameterSelectionStrategy.MULTI,
-    )
+    def default_configuration_impl(mode: Optional[str] = "MULTI"):
+        """Return the default test compilation configuration for a given strategy.
+
+        Args:
+            mode (Optional[str]): Whether we execute in a MONO or MULTI precision settings.
+
+        Returns:
+            The configuration.
+        """
+        if mode == "MULTI":
```

Comment:
really not a fan of this as it brings many changes in tests that I don't thing they are necessary. If it is absolutely necessary, please explain why and add a link to an issue here !

Reply:
Fixed